app/src/test/java/org/simple/clinic/contactpatient/ContactPatientUpdateTest.kt

```diff
@@ -479,6 +490,118 @@ class ContactPatientUpdateTest {
+        .overdueAppointmentLoaded(Just(overdueAppointment))
+        .removeAppointmentReasonSelected(Died)
+
+    spec
+        .given(model)
+        .whenEvent(RemoveAppointmentDoneClicked)
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(MarkPatientAsDead(patientUuid = patientUuid, appointmentUuid = appointmentUuid) as ContactPatientEffect)
+        ))
+  }
+
+  @Test
+  fun `when the appointment is removed because the patient is not responding, the appointment must be cancelled`() {
+    val model = defaultModel()
+        .patientProfileLoaded(patientProfile)
+        .overdueAppointmentLoaded(Just(overdueAppointment))
+        .removeAppointmentReasonSelected(NotResponding)
+
+    spec
+        .given(model)
+        .whenEvent(RemoveAppointmentDoneClicked)
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(CancelAppointment(appointmentUuid = appointmentUuid, reason = PatientNotResponding) as ContactPatientEffect)
+        ))
+  }
+
+  @Test
+  fun `when the appointment is removed because the phone number is invalid, the appointment must be cancelled`() {
+    val model = defaultModel()
+        .patientProfileLoaded(patientProfile)
+        .overdueAppointmentLoaded(Just(overdueAppointment))
+        .removeAppointmentReasonSelected(PhoneNumberNotWorking)
+
+    spec
+        .given(model)
+        .whenEvent(RemoveAppointmentDoneClicked)
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(CancelAppointment(appointmentUuid = appointmentUuid, reason = InvalidPhoneNumber) as ContactPatientEffect)
+        ))
+  }
+
+  @Test
+  fun `when the appointment is removed because the patient was transferred to another facility, the appointment must be cancelled`() {
+    val model = defaultModel()
+        .patientProfileLoaded(patientProfile)
+        .overdueAppointmentLoaded(Just(overdueAppointment))
+        .removeAppointmentReasonSelected(TransferredToAnotherFacility)
+
+    spec
+        .given(model)
+        .whenEvent(RemoveAppointmentDoneClicked)
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(CancelAppointment(appointmentUuid = appointmentUuid, reason = TransferredToAnotherPublicHospital) as ContactPatientEffect)
+        ))
+  }
+
+  @Test
+  fun `when the appointment is removed because the patient has moved to a private practitioner, the appointment must be cancelled`() {
+    val model = defaultModel()
+        .patientProfileLoaded(patientProfile)
+        .overdueAppointmentLoaded(Just(overdueAppointment))
+        .removeAppointmentReasonSelected(MovedToPrivatePractitioner)
+
+    spec
+        .given(model)
+        .whenEvent(RemoveAppointmentDoneClicked)
+        .then(assertThatNext(
+            hasNoModel(),
+            hasEffects(CancelAppointment(appointmentUuid = appointmentUuid, reason = AppointmentCancelReason.MovedToPrivatePractitioner) as ContactPatientEffect)
```

Comment:
Import the reason directly like the others?

Reply:
We cannot do this because we already have the `RemoveAppoinmentReason` sealed classes being imported which have a type named the same.